discovery.md

```diff
@@ -0,0 +1,378 @@
+  protocols. All producers MUST support at least one delivery protocol, and MAY
+  support additional protocols.
+- Constraints:
+  - REQUIRED
+  - MUST be non-empty.
+- Examples:
+  - ["HTTP"]
+  - ["HTTP", "AMQP", "KAFKA"]
+
+#### extensions
+
+- Type: `Map` of `String` to `String`
+- Description: Associative map of CloudEvents [Extension Content Attributes](https://github.com/cloudevents/spec/blob/master/spec.md#extension-context-attributes)
+  that are used for this event `type`. Keys MUST be confirm to the extension
+  content attributes naming rules and value are the type of the extension
+  attribute, confirming to the CloudEvents [type system](https://github.com/cloudevents/spec/blob/master/spec.md#type-system).
+- Constraints:
+  - OPTIONAL
+
+#### subscriptionendpoint
+
+- Type: `URI-reference`
+- Description: URI indicating where CloudSubscriptions `subscribe` API calls
+  should be sent to.
+- Constraints:
+  - REQUIRED
+
+#### subscriptionconfig
+
+- Type: `Map` of `String` to `String`
+- Description: A map indicating supported options for the `config` parameter
+  for the CloudSubscriptions subscribe() API call. Keys are the name of keys
+  in the allowed config map, the values indicate the type of that parameter,
+  confirming to the CloudEvents [type system](https://github.com/cloudevents/spec/blob/master/spec.md#type-system).
+  TODO: Needs resolution with CloudSubscriptions API
+- Constraints:
+  - OPTIONAL
+- Examples:
+
+#### subscriptionttl
+
+- Type: `Integer`
+- Description: Time (in seconds) after which subscriptions of this will will
+  automatically expire.
+- Constraints:
+  - OPTIONAL
+  - If present, must be a positive Integer >= 1.
+- Examples:
+  - `604800` - producer will expire subscriptions in 7 days
+
+#### authscope
+
+- Type: `String`
+- Description: Authorization scope required for creating subscriptions.
+  The actual meaning of this field is determined on a per-producer basis.
+- Constraints:
+  - OPTIONAL
+- Example:
+  - "storage.read"
+
+### Query API
+
+The query operation MUST be supported by compliant Event Producers. It is used
+by the Event Subscriber to get information about the events that the Event
+Producer produces. To retrieve the list of events produced by an Event Producer,
+the following data is to be provided. The query operation takes a key/value map
+(`config`) of type `string` to `string`.
+
+Each query parameter that is provided is an `and` operation. Each parameter
+MUST support exact match and prefix matching operations. Prefix matching uses
+the syntax of `prefix*` where the `*` character represents the end of the
+prefix.
+
+#### Query API keys
+
+##### producer
+
+- Type: `String`
+- Description: The name of the [producer](#producer) that is offering this
+  events.
+- Constraints:
+  - OPTIONAL
+  - If present, must be non-empty
+- Examples:
+  - "Github"
+  - "Awesome Cloud Storage"
+  - "Awesome Cloud\*"
+
+##### source
+
```

Comment:
Does this only apply to `source`, or is it matched against `sourcestructure`, too?

Reply:
just to source - `sourcestructure` is meant to be informational. 

I don't see a use case for including it in the query, but I could be missing something.

Reply:
Removing source for now, based on feedback from others.